cmd/controller/app/database/mongodb/code.go

```
@@ -202,3 +202,22 @@ func (db *MongoService) GetDesignCode(userId string, designId string, fileVer st
 
 	return util.ZipFile(updatedDoc.DataSet)
 }
+func (db *MongoService) DeleteDesignCode(userId string, designId string, version string) error {
+	zap.S().Debugf("delete design code 1: %v, %v,%v", userId, designId, version)
+
+	if version == "" || version == latestVersion {
```

Since it involves deletion of resource, we may want to be conservative. Perhaps would it be better to return an error if a version string is empty?

---

done.